### ISSUE TITLE
Create User Profile Component

### ISSUE THREAD

**Issue by jonathanprozzi** [HausDAO/daohaus-monorepo#101]
## Overview

Build out the Hub App Header Component that leverages Ceramic Basic Profile data.

![Image](https://user-images.githubusercontent.com/9438776/166733099-e5349d6f-3131-4b60-b708-f96c43ce9ddd.jpg)

- This is a crop of the Header -- the Figma link shows the Header in the full context of the Hub App UI
- The profile component is on the right of the Header

[Figma Link](https://www.figma.com/file/9lCwprCNaoAmUDmiMQpFEA/DAOhaus-v3?node-id=4185%3A39751)

## User Stories

As a user I want to see my Ceramic profile data when I'm connected to the Hub App so that I can see my Ceramic profile data and edit it.

## Fallback Data

Since Ceramic Basic Profile fields are optional, we'll want to use a set of fallbacks:

- Fields: user name, emoji, avatar, background image, bio/description
- Fallbacks (since Basic Profile fields are all optional):
  - avatar: blockie
  - username: truncated address
  - address: (no ens: address) (ens: ens)
  - show ENS regardless if they have an ENS
  - bio: blank space OR GPT-3 bio :D
  - emoji: blank
  - background image: app background image

## Scenarios

```gherkin
Feature: User Profile (Connected, has Profile)

Scenario: User has set up a Ceramic Profile and is connected to the Hub with the same address.
    Given that I'm on the Hub App landing page
    And I've connected my wallet
    And I have set up a Ceramic Profile for my connected address
    Then the Header Profile UI displays my Basic Profile info (username, address, avatar, bio, Edit Profile link)
```

```gherkin
Feature: User Profile (Connected, Ceramic Profile, No ENS)

Scenario: User has connected their wallet, has set up a Ceramic Profile, and is connected to the Hub with the same address but has NOT set up their ENS.
    Given that I'm on the Hub App landing page
    And I have a Ceramic Profile but not an ENS
    Then the Header Profile UI displays a truncated view of my Eth address
```

```gherkin
Feature: User Profile (Connected, No Ceramic Profile, has ENS)

Scenario: User has connected their wallet, hasn't set up a Ceramic Profile, but does have an ENS.
    Given that I'm on the Hub App landing page
    And I haven't set up my Ceramic Profile but I have an ENS
    Then the Header Profile UI displays the user's ENS but no Ceramic Profile data
```

```gherkin
Feature: User Profile (Connected, No Ceramic Profile)

Scenario: User has connected their wallet, but has not set up a Ceramic profile with the connected address.
    Given that I'm on the Hub App landing page
    And I've connected my wallet
    And I haven't set up a Ceramic Profile for my connected address
    Then the Header Profile UI shows a CTA explaining DAOhaus leverages Ceramic and a "Setup Ceramic Profile" Button to take them to self.id
```

```gherkin
Feature: User Profile (Connected, Has Ceramic Profile, has not set up Basic Profile)

Scenario: User has connected their wallet, have a Ceramic profile with the connected address, but haven't added their Basic Profile information.
    Given that I'm on the Hub App landing page
    And I've connected my wallet
    And I have a Ceramic Profile for my connected address but haven't added any Basic Profile information
    Then the Header Profile UI displays the fallback/placeholders for the Basic Profile schema fields
```

## Related Issues

- The data fetching for this component leverages #92 
- Some of the functionality for the UI is similar to #91 since the "Connect Wallet" pulls from Ceramic (if available)

## Deliverables

- Hub App Header UI pulling in data from Ceramic with the fallbacks for the optional fields

**Comment by jonathanprozzi** [comment]
One other item we discussed was potentially storing any user preferences in Ceramic as well:

- Store user preferences in a DAOhaus Ceramic schema
  - Edit Preferences
  - If they don’t have a profile set, we’d create a DID for them
  - Update, but can’t make it destructive
  - Updating and adding is fine, but removing is not

**Comment by alexkeating** [comment]
@skuhlmann @jonathanprozzi @earth2travis following up on whether we have a link to the ENS profile to use

**Comment by jonathanprozzi** [comment]
Closed by #145

**Comment by jonathanprozzi** [comment]
Closed by #145 🎉
